ctowasm/src/modules/util.ts

```diff
@@ -100,3 +102,64 @@ export function printSharedGlobalVariables(
     console.log(`${name}: ${value.value}`);
   }
 }
+
+/**
+ * Store a Foreign JS Object in memory by attaching a unique identifier and return it to C Side
+ */
+export function storeObjectInMemoryAndRegistry(
+    memory: WebAssembly.Memory,
+    objectReferenceRegistry: Map<number, Object>,
+    sharedWasmGlobalVariables: SharedWasmGlobalVariables,
+    allocatedBlocks: Map<number, number>,
+    freeList: any[],
+    obj: Object,
+) {
+    if (!obj) {
+        throw Error("Cannot store null object in memory");
+    }
+    const objIdentifier = new Uint8Array([
+      (FOREIGN_OBJ_IDENTIFIER >> 8) & 0xFF,  // High byte (0xF0)
+      FOREIGN_OBJ_IDENTIFIER & 0xFF          // Low byte (0xBA)
```

Comment:
what is the purpose of FOREIGN_OBJ_IDENTIFIER ?  it does not seem to be used anywhere. It seems you're just using it as a placeholder value for the address that you give your foreign object for the value of its `void *`. In that case you could just leave it as zeros since from the C code deferencing the `void *` would be UB. Unless you need it on the JS side, which dosent seem to be the case in this PR.

Reply:
Alternatively the C standard does not specify that the value of a pointer must be an actual address in memory, merely a "reference". In that case you could instead come up with a solution to giving a unique value for your foreign object `void *` variables which are guaranteed to not be equal to any other foreign object or actual C object or NULL. To do this, you could reserve a portion of the addressable memory range beforehand and assign foreign object `void *` values from there, or even just increase the underlying primitive type of pointers from `unsigned int` to `unsigned long int` (32bit to 64bit), while the WASM addressable space is still 32bit (limitation of WASM memory addressing AFAIK). You could modify this in _c-slang/ctowasm/src/common/constants.ts_ (might be some other changes needed too). That way you can just assign your foreign object `void *` values from outside the unsigned 32bit numeric range.